### PR TITLE
feat: Update Rust builder, add `target` flag

### PR DESCRIPTION
**Pull request by stevenj**
# Description

Updated rust builder, add new configuration options for building step. Replaced bash scripts with python one.

## Related Issue(s)

Part of the https://github.com/input-output-hk/hermes/issues/20

## Description of Changes

- replaced `std_build.sh` and `std_check.sh` with Python implementation `std_check.py` and `std_build.py`.
- modified cargo `config.toml` files, provided new `target.wasm32-unknown-unknown` compile options, removed `--locked` flag from all aliases (because we are going to remove `Cargo.lock` files from the repo), disabled `cargo` progress bar and disabled `verbose`.
- Moved `SMOKE_TEST` udc functionality to the `std_build.py` script.
- removed `nushell` installation.
- some other cleanups

### CODE REVIEW

File: earthly/rust/scripts/std_build.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    args = parser.parse_args()
+
+    results = cli.Results("Rust build")
+
+    target_flag = ""
+    if args.target != "":
+        target_flag += f" --target={args.target} "
+
+    # Build the code.
+    results.add(cli.run("cargo build " \
```
Comment (stevenj) [inline earthly/rust/scripts/std_build.py:33]:
```suggestion
    results.add(cli.run("cargo build "
```

File: earthly/rust/scripts/std_build.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                        f"{target_flag} " \
```
Comment (stevenj) [inline earthly/rust/scripts/std_build.py:34]:
```suggestion
                        + f"{target_flag} "
```

File: earthly/rust/scripts/std_build.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                        "--release ",
```
Comment (stevenj) [inline earthly/rust/scripts/std_build.py:35]:
```suggestion
                        + "--release ",
```

Comment (stevenj) [reply earthly/rust/scripts/std_build.py:35]: Don't suppress line breaks, join strings using `+` across multiple lines.
This applies to all instances of this in the PR

**Comment by Mr-Leshiy** [comment]
updated docs:
https://input-output-hk.github.io/catalyst-ci/branch/feat_rust_wasm/guides/languages/rust/

**Review by Mr-Leshiy — APPROVED** [review]
LGTM

**Pull request merged**